packages/medusa-file-s3/src/services/s3.js

```diff
@@ -19,9 +19,19 @@ class S3Service extends AbstractFileService {
   upload(file) {
     this.updateAwsConfig()
 
+    return this.uploadFile(file, "public-read")
```

Comment:
thought: It seems the interface is used slighly differently from plugin to plugin. Here we use it to directly pass the access level while in Minio it's a flag. Would it make sense to consistently use `uploadFile` with a flag to indicate the privacy level?

This would mean, that we'd have to have the following in S3:
```
uploadFile(file, makePrivate) {
    const s3 = new aws.S3()
    const params = {
      ACL: makePrivate ? "private" : "public-read",
      ...
```

And the following in Minio:
```
uploadFile(file, makePrivate) {
    const parsedFilename = parse(file.originalname)
    const fileKey = `${parsedFilename.name}-${Date.now()}${parsedFilename.ext}`

    const s3 = new aws.S3()
    const params = {
      ACL: makePrivate ? "private" : "public-read",
      Bucket: makePrivate ? this.private_bucket_ : this.bucket_,
      ....
```

And so forth for all plugins. 

Let me know what you think.

Reply:
pretty agree on that in order to make the api consistent and easier to use. Would just say that the second argument could be like:
```typescript
uploadFile(file, options: { isProtected?: boolean } = { isProtected: false })
```

Reply:
I like that. This would also allow us to add an ACL-specific prop in case users need more granular access control:
```
uploadFile(file, options: { isProtected?: boolean, acl?: string } = { isProtected: false })
```